ddtrace/tracer/rules_sampler.go

```diff
@@ -54,6 +54,55 @@ func (r *rulesSampler) HasSpanRules() bool { return r.spans.enabled() }
 
 func (r *rulesSampler) TraceRateLimit() (float64, bool) { return r.traces.limit() }
 
+type Provenance int32
+
+const (
+	Local    Provenance = iota
+	Customer Provenance = 1
+	Dynamic  Provenance = 2
+)
+
+var (
+	ProvenanceName = map[Provenance]string{
+		Local:    "local",
+		Customer: "customer",
+		Dynamic:  "dynamic",
+	}
+	ProvenanceValue = map[string]Provenance{
+		"local":    Local,
+		"customer": Customer,
+		"dynamic":  Dynamic,
+	}
+)
```

Comment:
This might be quite error prone, 
```suggestion
var	Provenances = []Provenance{Local, Customer, Dynamic}

func (p Provenance) String() string {
	switch p {
	case Local:
		return "local"
	case Customer:
		return "customer"
	case Dynamic:
		return "dynamic"
	default:
		return ""
	}
}
```

Reply:
we still need the definition of 
```const (
	Local    Provenance = iota
	Customer Provenance = 1
	Dynamic  Provenance = 2
)
```

Also, the maps are referred in the tests.

Reply:
THe tests should be changed accordingly. The iota is fine to keep, that I selected by accident.